hypothesis-python/src/hypothesis/database.py

```diff
@@ -46,7 +55,15 @@
 if TYPE_CHECKING:
     from typing import TypeAlias
 
+    from watchdog.observers.api import BaseObserver
+
 StrPathT: "TypeAlias" = Union[str, PathLike[str]]
+ListenerEventType: "TypeAlias" = Literal["save", "delete", "move"]
+SaveDataT: "TypeAlias" = tuple[bytes, bytes]  # key, value
+DeleteDataT: "TypeAlias" = tuple[bytes, Optional[bytes]]  # key, value
+MoveDataT: "TypeAlias" = tuple[bytes, bytes, bytes]  # src, dest, value
+ListenerDataT: "TypeAlias" = Union[SaveDataT, DeleteDataT, MoveDataT]
+ListenerT: "TypeAlias" = Callable[[ListenerEventType, ListenerDataT], Any]
```

Comment:
```suggestion
ListenerT: "TypeAlias" = Callable[[tuple[ListenerEventType, ListenerDataT]], None]
```
Taking a single argument and unpacking later makes it easier to hook up many listeners, including e.g. [`channel.send_nowait`](https://trio.readthedocs.io/en/stable/reference-core.html#trio.MemorySendChannel.send_nowait).

Reply:
`Any` return is intentional; we don't rely on the listener returning None, so we do actually want to allow listeners with return values.